### DISCUSSION TITLE
a question about “Anistropic mobility” or Interface Orientation

### DISCUSSION THREAD

**Discussion by dongcongcong606** [idaholab/moose#18016] [Q&A Modules: Phase field] [answered]
Dear friends！
Hi everyone, I have a question about the code of "anistropy". I would like to know how to introduce anisotropic mobility into the phase field model in Moose. And I read some file  such as "InterfaceOrienation.c" 、"ADInterfaceOrienationMaterial.c" . But I don't know which of these two models is a better model for anisotropic mobility. And why the "Orientation angle" in terms of acos rather than atan? I am very grateful to all the friends and experts present here for giving me directions. Thanks again!

**Comment by laagesen** [comment, marked as answer]
Hello- the `InterfaceOrientationMaterial` is to account for anisotropic interfacial energy rather than anisotropic order parameter mobility. So, you do not want to use this material if you want to include anisotropic mobility in the Allen-Cahn equation. If you want a similar 2-D anisotropy as described in the Kobayashi paper, you would need to create a new material that would be very similar to `InterfaceOrientationMaterial`. However, rather than `eps`, you would want to create a material property called `L` (for example) that could be supplied to the `mob_name` parameter in the `AllenCahn` and `ACInterface` kernels (also make sure to also specify `variable_L = true` to `ACInterface`).

**Reply by dongcongcong606** [reply]
Thank you very much. I obey your suggestion to define a new material named "ADANisotropicFirst"  in the following file, and I also introduce an anisotropic function to represent "mobility" ，and whether I am supposed to adjust or even define a new kernel like “Kobayashi1” to describe my own Jacobian codes ？Please forgive me for my shallow knowledge. I would like to express my gratitude again here.

**Reply by laagesen** [reply]
If you use the AD versions of the other kernels, you won't need to code the Jacobians, they will be calculated automatically.

**Reply by dongcongcong606** [reply]
> If you use the AD versions of the other kernels, you won't need to code the Jacobians, they will be calculated automatically.
thank you very much
